dropwizard-client/src/test/java/io/dropwizard/client/DropwizardApacheConnectorTest.java

```diff
@@ -0,0 +1,180 @@
+import org.junit.rules.ExpectedException;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import java.net.SocketTimeoutException;
+import java.net.URI;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.CoreMatchers.any;
+
+public class DropwizardApacheConnectorTest {
+    private static final int SLEEP_TIME_IN_MILLIS = 500;
+    private static final int DEFAULT_CONNECT_TIMEOUT = 200;
+    private static final String NON_ROUTABLE_IP_ADDRESS = "10.255.255.1";
+
+    @ClassRule
+    public static DropwizardAppRule<Configuration> APP_RULE =
+            new DropwizardAppRule<>(TestApplication.class, Resources.getResource("yaml/dropwizardApacheConnectorTest.yml").getPath());
+    public final URI TEST_URI = java.net.URI.create("http://localhost:" + APP_RULE.getLocalPort());
+    public final URI URI_THAT_CANNOT_BE_CONNECTED_TO = java.net.URI.create("http://" + NON_ROUTABLE_IP_ADDRESS);
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    private JerseyClientConfiguration clientConfiguration;
+
+    @Before
+    public void setup() {
+        clientConfiguration = new JerseyClientConfiguration();
+        clientConfiguration.setConnectionTimeout(Duration.milliseconds(SLEEP_TIME_IN_MILLIS / 2));
+        clientConfiguration.setTimeout(Duration.milliseconds(DEFAULT_CONNECT_TIMEOUT));
+    }
+
+    private Client createStandardClient() {
+        return new JerseyClientBuilder(new MetricRegistry())
+                .using(Executors.newSingleThreadExecutor(), Jackson.newObjectMapper())
+                .using(clientConfiguration)
+                .build("test'");
+    }
+
+
+    @Test
+    public void when_no_read_timeout_override_then_client_request_times_out() {
+        thrown.expect(ProcessingException.class);
+        thrown.expectCause(any(SocketTimeoutException.class));
+
+        Client client = createStandardClient();
+
+        client.target(TEST_URI + "/long_running")
+                .request()
+                .get();
+    }
+
+    @Test
+    public void when_read_timeout_override_created_then_client_requests_completes_successfully() {
+        Client client = createStandardClient();
+
+        client.target(TEST_URI + "/long_running")
+                .property(ClientProperties.READ_TIMEOUT, SLEEP_TIME_IN_MILLIS * 2)
+                .request()
+                .get();
+    }
+
+    @Test
+    public void connect_timeout_override_changes_how_long_it_takes_for_a_connection_to_timeout() {
+        Client client = createStandardClient();
+
+        // before override
+        assertThatGetConnectonTimeoutForTarget(
+                client.target(URI_THAT_CANNOT_BE_CONNECTED_TO)
+        ).isLessThan(DEFAULT_CONNECT_TIMEOUT + 300);
```

Comment:
Sorry, I don't understand where number 300 is taken. Is there some dependency?
Also, shouldn't this assertion be `isGreaterThan(DEFAULT_CONNECT_TIMEOUT);`?